faunadb-android/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -300,18 +299,15 @@ private void handleQueryErrors(Response response) throws FaunaException {
     }
   }
 
-  private <V> ListenableFuture<V> handleNetworkExceptions(ListenableFuture<V> f) {
-    ListenableFuture<V> f1 = Futures.catching(f, ConnectException.class, new Function<ConnectException, V>() {
-      @Override
-      public V apply(ConnectException input) {
-        throw new UnavailableException(input.getMessage());
-      }
-    });
-
-    return Futures.catching(f1, TimeoutException.class, new Function<TimeoutException, V>() {
-      @Override
-      public V apply(TimeoutException input) {
-        throw new UnavailableException(input.getMessage());
+  private <V> CompletableFuture<V> handleNetworkExceptions(CompletableFuture<V> f) {
+    return f.whenComplete(new BiConsumer<V, Throwable>() {
```

Comment:
if you use `f.exceptionally()` you wouldn't need to create a `BiConsumer`, just a `Function`

Reply:
I hadn't discovered `exceptionally()` yet, when I wrote this. Good idea. :)

Reply:
Oh, I see, case the exception wasn't `ConnectException` or `TimeoutException` it should just ignore the exception, but is it right? Not sure now.

Reply:
That seems to break the semantics of this function, though. Hmm.